src/metabase/collections/api.clj

```diff
@@ -1207,6 +1203,15 @@
                                   :official-collections-first? (or (nil? official_collections_first)
                                                                    (boolean official_collections_first))}})))
 
+;;; ------------------------------------------ Fetching a single Collection -------------------------------------------
+
+(api.macros/defendpoint :get "/:id"
+  "Fetch a specific Collection with standard details added"
+  [{:keys [id]} :- [:map
+                    [:id [:or ms/PositiveInt :string]]]]
+  (let [resolved-id (eid-translation/->id-or-404 :collection id)]
```

Comment:
`id-or-404` makes it return 404 errors when the entity id lookup fails; this is intended for use with endpoints only. feel free to recommend a better name.